client/src/components/EditablePolyline.tsx

```diff
@@ -3,39 +3,149 @@ import { Polyline as CorePolyline, LeafletEvent, LatLng } from 'leaflet';
 import { Polyline, PolylineProps } from 'react-leaflet';
 import { omit } from 'lodash';
 
-import { Unit, Point } from '../models';
+import { Unit, Point, AppStateContainer } from '../models';
 import { gameService } from '../services';
 
 export type EditablePolylineProps = PolylineProps & {
   unit: Unit;
 };
 
+type AppStateReturnType = ReturnType<typeof AppStateContainer.useContainer>;
+
+function redrawUnit(unit: Unit, appState: AppStateReturnType) {
```

Comment:
Could you explain to me what is happening here? I'm not following the logic here.

Reply:
useAppState() creates a new type by extending AppState which contains updateUnit which is needed by redrawUnit so we need the type returned by the useContainer function(=return type of useAppState).

Reply:
Sorry, I meant I didn't understand what is happening inside the function redrawUnit. But after going to see where this was called, it looks like the line is redrawn when the user make an invalid action. I think this is fine for now, but could be revisited as we get further along.